classes/task/ReviewReminder.inc.php

```diff
@@ -0,0 +1,167 @@
+<?php
+
+/**
+ * @file classes/task/ReviewReminder.inc.php
+ *
+ * Copyright (c) 2013-2016 Simon Fraser University Library
+ * Copyright (c) 2003-2016 John Willinsky
+ * Distributed under the GNU GPL v2. For full terms see the file docs/COPYING.
+ *
+ * @class ReviewReminder
+ * @ingroup tasks
+ *
+ * @brief Class to perform automated reminders for reviewers.
+ */
+
+import('lib.pkp.classes.scheduledTask.ScheduledTask');
+
+define('REVIEW_REMIND_AUTO', 'REVIEW_REMIND_AUTO');
+define('REVIEW_REQUEST_REMIND_AUTO', 'REVIEW_REQUEST_REMIND_AUTO');
+
+class ReviewReminder extends ScheduledTask {
+
+	/**
+	 * Constructor.
+	 */
+	function ReviewReminder() {
+		parent::ScheduledTask();
+	}
+
+	/**
+	 * @copydoc ScheduledTask::getName()
+	 */
+	function getName() {
+		return __('admin.scheduledTask.reviewReminder');
+	}
+
+	function sendReminder ($reviewAssignment, $submission, $context, $reminderType = REVIEW_REMIND_AUTO) {
+		$reviewAssignmentDao = DAORegistry::getDAO('ReviewAssignmentDAO');
+		$userDao = DAORegistry::getDAO('UserDAO');
+		$reviewId = $reviewAssignment->getId();
+
+		$reviewer = $userDao->getById($reviewAssignment->getReviewerId());
+		if (!isset($reviewer)) return false;
+
+		import('lib.pkp.classes.mail.SubmissionMailTemplate');
+
+		$reviewerAccessKeysEnabled = $context->getSetting('reviewerAccessKeysEnabled');
+
+		$email = new SubmissionMailTemplate($submission, $reviewerAccessKeysEnabled ? $reminderType . '_ONECLICK' : $reminderType, $context->getPrimaryLocale(), $context, false);
```

Comment:
Rather than assembling a string, I'd suggest something like...

    switch(true) {
        case $reviewer->getAccessKeysEnabled() && $reminderType == REVIEW_REMIND_AUTO && (...etc...):
            $templateName = '(template name here)';
            break;
        case ...:
            ...etc...
    }

The trouble with assembling symbolic names with string functions is that you can't grep the codebase for it.